### ISSUE TITLE
Example for documents that works for all versions

### ISSUE THREAD

**Issue by DavidKmn** [anggrayudi/SimpleStorage#9]
I have found working with download/document types a bit harder than with normal media type files. Since some methods are market with Q > for download functions. I have wrote the below as a workaround helper functions I think could also help others with this, what do you think? @anggrayudi 

```

 fun createDocument(context: Context, file: FileDescription): DocumentFile? {
        var subFolder = ""
        documentsSubfolder?.let {
            subFolder = "${it}/${file.subFolder}"
        } ?: run {
            subFolder = file.subFolder
        }
        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
            MediaStoreCompat.createDownload(context, file = com.anggrayudi.storage.media.FileDescription(file.name, subFolder, file.mimeType))?.toDocumentFile()
        } else {
            subFolder += "/${file.name}"
            DocumentFileCompat.createFile(context, basePath = subFolder, mimeType = file.mimeType)
        }
    }

```
```
    fun fetchDocument(context: Context, subFolder: String = "", name: String): DocumentFile? {
        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
            MediaStoreCompat.fromFileName(context, MediaType.DOWNLOADS, name)?.toDocumentFile()
        } else {
            var basePath = ""
            documentsSubfolder?.let {
                basePath = "${it}/${subFolder}"
            } ?: run {
                basePath = subFolder
            }
            basePath += "/${name}"
            DocumentFileCompat.fromSimplePath(context, basePath)
        }
    }
```

**Comment by anggrayudi** [comment]
1. `fun createDocument()` capability was added since `0.4.0` as `DocumentFileCompat#createDownloadWithMediaStoreFallback()`.
2. `toDocumentFile()` is not reliable when you have no URI permission, thus I will not add fun `fetchDocument()`.

**Comment by DavidKmn** [comment]
Thanks for your help. 
1. Is there a fetch for a download that works on all versions?
2. Is there a fetch to list all downloads created (using fromMediaType we can only choose downloads if > version Q), what would we use for other versions?

**Comment by anggrayudi** [comment]
For Android Q & you have no URI permission, use `MediaStoreCompat.fromMediaType(context, MediaType.DOWNLOADS)`

If you have URI permission, regardless of API level, better use `DocumentFileCompat.fromPublicFolder(context, PublicDirectory.DOWNLOADS)?.listFiles()`

**Comment by DavidKmn** [comment]
What about fetching a single download by name for < android Q?

**Comment by anggrayudi** [comment]
You can use `DocumentFileCompat.fromPublicFolder(context, PublicDirectory.DOWNLOADS)?.findFile(filename)`

**Comment by DavidKmn** [comment]
Okay great 😊 
Is there something like File.lastModified() to get the last modification date for a file/media item. For example to display them in a sorted manner in an file picker display?

**Comment by DavidKmn** [comment]
> 1. `fun createDocument()` capability was added since `0.4.0` as `DocumentFileCompat#createDownloadWithMediaStoreFallback()`.
> 2. `toDocumentFile()` is not reliable when you have no URI permission, thus I will not add `fun fetchDocument()`.

Digging deeper I see a problem with createDownloadWithMediaStoreFallback, it returns a Uri whilst putting isPending flag to true. There is no way to set isPending to false (MediaFile and DocumentFile on the other hand do have isPending setter) once we have written to the created file, this in turn means that the file will not be readable.

**Comment by anggrayudi** [comment]
I've released [0.4.3](https://github.com/anggrayudi/SimpleStorage/releases/tag/0.4.3). Please read the [release notes](https://github.com/anggrayudi/SimpleStorage/releases/tag/0.4.3).

**Comment by DavidKmn** [comment]
Thanks for the release, I see isPending is no longer set by default. Maybe the documentation should be updated to let the others know that isPending on the MediaFile has to be manually set to true when writing to the file & set to false once done writing to the file?
